images/virtualization-artifact/pkg/controller/vmip/internal/iplease_handler.go

```diff
@@ -122,11 +124,37 @@ func (h IPLeaseHandler) createNewLease(ctx context.Context, state state.VMIPStat
 		vmipStatus.Address = vmip.Spec.StaticIP
 	}
 
-	if !h.ipService.IsAvailableAddress(vmipStatus.Address, state.AllocatedIPs()) {
-		msg := fmt.Sprintf("VirtualMachineIP cannot be created: the address %s has already been allocated for another vmip", vmip.Spec.StaticIP)
+	err := h.ipService.IsAvailableAddress(vmipStatus.Address, state.AllocatedIPs())
+	if err != nil {
+		vmipStatus.Address = ""
+		msg := fmt.Sprintf("the VirtualMachineIP cannot be created: %s", err.Error())
 		h.logger.Info(msg)
-		h.recorder.Event(vmip, corev1.EventTypeWarning, vmipcondition.VirtualMachineIPAddressLeaseLost, msg)
-		return reconcile.Result{}, nil
+
+		mgr := conditions.NewManager(vmipStatus.Conditions)
+		conditionBound := conditions.NewConditionBuilder(vmipcondition.BoundType).
+			Generation(vmip.GetGeneration())
+
+		switch {
+		case errors.Is(err, service.ErrIPAddressOutOfRange):
+			vmipStatus.Phase = virtv2.VirtualMachineIPAddressPhasePending
+			mgr.Update(conditionBound.Status(metav1.ConditionFalse).
+				Reason(vmipcondition.VirtualMachineIPAddressIsOutOfTheValidRange).
+				Message(fmt.Sprintf("The requested address %s is out of the valid range",
+					vmip.Spec.StaticIP)).
+				Condition())
+			h.recorder.Event(vmip, corev1.EventTypeWarning, vmipcondition.VirtualMachineIPAddressIsOutOfTheValidRange, msg)
+		case errors.Is(err, service.ErrIPAddressAlreadyExist):
+			vmipStatus.Phase = virtv2.VirtualMachineIPAddressPhasePending
+			mgr.Update(conditionBound.Status(metav1.ConditionFalse).
+				Reason(vmipcondition.VirtualMachineIPAddressLeaseAlready).
+				Message(fmt.Sprintf("VirtualMachineIPAddressLease %s is bound to another VirtualMachineIP",
+					common.IpToLeaseName(vmipStatus.Address))).
+				Condition())
+			h.recorder.Event(vmip, corev1.EventTypeWarning, vmipcondition.VirtualMachineIPAddressLeaseAlready, msg)
+		}
+
+		vmipStatus.Conditions = mgr.Generate()
+		return reconcile.Result{}, err
```

Comment:
Is it OK to reconcile on ErrIPAddressOutOfRange or ErrIPAddressAlreadyExist errors? They seem a final state for resource and we can `return reconcile.Result{}, nil` after updating status.

Reply:
Indeed, he did so. 
Further reconciles will lead to nothing.